cpp/src/arrow/acero/aggregate_benchmark.cc

```diff
@@ -866,5 +890,47 @@ BENCHMARK(TDigestKernelDoubleMedian)->Apply(QuantileKernelArgs);
 BENCHMARK(TDigestKernelDoubleDeciles)->Apply(QuantileKernelArgs);
 BENCHMARK(TDigestKernelDoubleCentiles)->Apply(QuantileKernelArgs);
 
+//
+// RowSegmenter
+//
+
+template <typename... Args>
+static void BenchmarkRowSegmenter(benchmark::State& state, Args&&...) {
+  int64_t num_rows = state.range(0);
+  int64_t num_segments = state.range(1);
+  ASSERT_NE(num_segments, 0);
+  ASSERT_GE(num_rows, num_segments);
+  int64_t num_segment_keys = state.range(2);
+  // Adjust num_rows to be a multiple of num_segments.
+  num_rows = num_rows / num_segments * num_segments;
+
+  // A trivial column to count from.
+  auto arg = ConstantArrayGenerator::Zeroes(num_rows, int64());
+  // num_segments segments, each having identical num_rows / num_segments rows of the
+  // associated segment id.
+  ArrayVector segments(num_segments);
+  for (int i = 0; i < num_segments; ++i) {
+    ASSERT_OK_AND_ASSIGN(
+        segments[i],
+        Constant(std::make_shared<Int64Scalar>(i))->Generate(num_rows / num_segments));
+  }
+  // Concat all segments to form the segment key.
+  ASSERT_OK_AND_ASSIGN(auto segment_key, Concatenate(segments));
+  // num_segment_keys copies of the segment key.
+  ArrayVector segment_keys(num_segment_keys, segment_key);
```

Comment:
Less trivially, you can, for each segmented key, 1) generate a random Int64 array of the right cardinality (using the min and max values) 2) sort it to make it segmented. Then the multi-column segments will not trivially map to the individual key segments.

Reply:
> Then the multi-column segments will not trivially map to the individual key segments.

Sorry, I don't follow this part :(

Could you elaborate a bit? Thank you.

Reply:
I mean that you have the exact same segment lengths in each segment key column. By using random generation the patterns would be slightly different.

Reply:
Ah, you mean introducing some randomness into the individual segment lengths. That sounds good given that this particular benchmark needs to be more realistic.

Will do. Thank you for elaborating.

Reply:
Done.